### DISCUSSION TITLE
C API: All methods to get query result data are deprecated.

### DISCUSSION THREAD

**Discussion by vbmithr** [duckdb/duckdb#12881] [Q&A] [answered]
I'm just a bit confused here, it seems that `duckdb_result_get_chunk`, supposedly the newest and best way to get result data, is itself deprecated even though other deprecated functions are deprecated in favour of this.

**Comment by Tishj** [comment, marked as answer]
The newest method to get a chunk from a query result is `duckdb_fetch_chunk`, we will update the header to make this clear

The reason that `duckdb_result_get_chunk` is being deprecated is because the QueryResult interface doesnt support this, only certain implementations have the ability to fetch a specific chunk index (MaterializedQueryResult)
